components/insight/SRC/org/openmicroscopy/shoola/util/ui/UIUtilities.java

```diff
@@ -2935,4 +2937,30 @@ public static String replaceWindowsPathSeparator(String path) {
     public static String replaceNonWordCharacters(String name) {
         return name.replaceAll("\\W", "_");
     }
+    
+    /**
+     * As workaround for https://bugzilla.gnome.org/show_bug.cgi?id=759492 an
+     * internal repaint has to be triggered for undecorated frames in order to
+     * get displayed correctly. This is done by changing the size of the frame
+     * (making it one pixel bigger); a simple call to repaint() is not
+     * sufficient.
+     * 
+     * @param w
+     *            The {@link Window}
+     */
+    public static void applyGnome3Workaround(Window w) {
+        boolean undecorated = false;
+        if (w instanceof Frame)
+            undecorated = ((Frame) w).isUndecorated();
+        if (w instanceof Dialog)
+            undecorated = ((Dialog) w).isUndecorated();
+
+        if (undecorated) {
+            String desktop = System.getenv("XDG_CURRENT_DESKTOP");
```

Comment:
Would you expect this value to ever be changed at runtime? If not, perhaps:

```
private final static boolean GNOME;
static {
    String desktop = System.getenv("XDG_CURRENT_DESKTOP");
    GNOME = desktop != null && desktop.toLowerCase().contains("gnome");
}
```

followed by:

```
public static void applyGnome3Workaround(Window w) {
    if (!GNOME) return;
```